src/autoschedulers/anderson2021/GPULoopInfo.h

```diff
@@ -42,9 +42,23 @@ struct GPULoopInfo {
 
     std::vector<int64_t> get_inner_serial_loop_extents(const LoopNest *loop_nest) const;
 
-    std::unique_ptr<ThreadInfo> create_thread_info();
+    // assert-fails if create_thread_info() has *already* been called.
+    const ThreadInfo *create_thread_info();
+
+    // Note: if create_thread_info() has not been called yet, this will return nullptr.
+    // TODO: should this just assert-fail if thread_info is null? I don't know if
+    // downstream code might legitimately check for null to mean "no thread_info exists".
+    const ThreadInfo *get_thread_info() const {
```

Comment:
This can legitimately return null for scalars after https://github.com/halide/Halide/pull/7726 (e.g. [here](https://github.com/halide/Halide/blob/main/src/autoschedulers/anderson2021/LoopNest.cpp#L978)) so I don't think it should assert-fail.

Reply:
Comment added.